### PR TITLE
Don't show No Results on select empty state

### PR DESCRIPTION
**Pull request by addepar-andy**
If select component has no content, then don't show the "No results match" result.

Reviewers:
@Addepar/ui2-developers

### CODE REVIEW

**Comment by igillis** [comment]
lgtm

**Comment by cyril-sf** [comment]
it needs a test

**Comment by sicasli** [comment]
I added a view for when the content is empty. This view can be used to show a help message about why the content is empty.

File: src/templates/select.hbs [hunk 1]
```diff
@@ -30,8 +30,16 @@
     {{#if isLoading}}
       <span class="ember-select-loading">Loading...</span>
     {{/if}}
-    {{#if hasNoResults}}
-      <span class="ember-select-no-results">No results match "{{query}}"</span>
+    {{#if contentIsEmpty}}
+      <div class="ember-select-empty-content">
+        {{#if emptyContentView}}
```
Comment (cyril-sf) [inline src/templates/select.hbs]: this condition is not tested

File: tests/unit/select_test.coffee [hunk 2]
```diff
@@ -197,3 +199,42 @@ test 'Test query matching', ->
...
+    optionValuePath: 'code'
+    classNames: 'select-class-name'
+
+  @append()
+
+  selectElement = select.$()
+  openDropdown(selectElement)
+  andThen ->
+    ok isPresent(emptyContentSelector, selectElement),
+      'Empty content view displayed'
```
Comment (cyril-sf) [inline tests/unit/select_test.coffee]: this assertion is a bit misleading as it doesn't clearly test `emptyContentView`

File: src/templates/select.hbs (hunk 1 above)
Comment (sicasli) [inline src/templates/select.hbs]: done

Comment (cyril-sf) [inline src/templates/select.hbs]: Now you only tested that code path :smile: , we need to test both!

Comment (cyril-sf) [inline src/templates/select.hbs]: You can start your test the way it was before, then set the `emptyContentView` and check after that it's rendered

Comment (sicasli) [inline src/templates/select.hbs]: Yeah I thought it's possible to render empty view so the test might be useless. Okay updated.

File: tests/unit/select_test.coffee [hunk 3]
```diff
@@ -197,3 +199,55 @@ test 'Test query matching', ->
...
+      'Empty content block displayed'
+    ok isNotPresent('.empty-content-view', selectElement),
+      'Empty content view not displayed before specified'
+    ok isNotPresent(noResultSelector, selectElement),
+      '"No result" message not displayed'
+  andThen ->
+    Ember.run ->
+      select.set 'emptyContentView', EmptyContentView
+  andThen ->
+    ok isPresent('.empty-content-view', selectElement),
```
Comment (cyril-sf) [inline tests/unit/select_test.coffee]: And also

**Comment by cyril-sf** [comment]
LGTM after remaining comment and tests pass

File: tests/unit/select_test.coffee (hunk 3 above)
Comment (sicasli) [inline tests/unit/select_test.coffee]: wait what? `emptyContentView` is inside `emptyContentSelector`

Comment (cyril-sf) [inline tests/unit/select_test.coffee]: I meant `empty-content-view`

Comment (sicasli) [inline tests/unit/select_test.coffee]: Not sure if I understand what you mean here. `emptyContentSelector` and `empty-content-view` are both present at this point.

Comment (cyril-sf) [inline tests/unit/select_test.coffee]: You need to make sure that setting `emptyContentView` is going to remove that view from the DOM.

Comment (sicasli) [inline tests/unit/select_test.coffee]: Oh okay I see what you mean

**Pull request merged**